src/controllers/shared.controllers/director.name.controller.ts

```diff
@@ -35,8 +39,21 @@ export const postDirectorName = async (req: Request, res: Response, next: NextFu
   try {
     const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
     const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
+    const companyNumber= urlUtils.getCompanyNumberFromRequestParams(req);
     const session: Session = req.session as Session;
 
+    // Validating officer etag
+    const currentOfficer = await getOfficerFiling(session, transactionId, submissionId);
+    const appointmentId = currentOfficer.referenceAppointmentId as string;
+    const companyAppointment: CompanyAppointment = await getCompanyAppointmentFullRecord(session, companyNumber, appointmentId);
+    const doEtagsMatch = currentOfficer.referenceEtag === companyAppointment.etag ? true : false;
+    
+    if (!doEtagsMatch) {
```

Comment:
This method will be called for AP01 as well from what I can see? I don't think we'd want that if so

Reply:
It will yes. I'll add a condition to specify what flow it's in to stop it being called for AP01

Reply:
minor - Would it be cleaner to put it in the update controller itself rather than in the shared controller with a condition?

Reply:
Could be yeah. I'll keep the validation for the etag within the update controller